base/initdefs.jl

```diff
@@ -136,10 +136,10 @@ function init_load_path()
     end
     project = (JLOptions().project != C_NULL ?
         unsafe_string(Base.JLOptions().project) :
-        get(ENV, "JULIA_PROJECT", nothing))
+        get(ENV, "JULIA_PROJECT", ""))
     HOME_PROJECT[] =
         project == "" ? nothing :
-        project == "@." ? current_project() : project
+        project == "@." ? current_project() : abspath(project)
```

Comment:
If we resolve this at startup time, you can't do `--project=.` and have it follow you around. I'm not sure if resolving it at startup or having it follow you around is better.

Reply:
Following around to the exact `pwd` is something I think you rarely want. 

```
➜  Pkg git:(kc/free_repo) ✗ ~/julia/julia --project=.
(Pkg) pkg>

shell> cd src/
/Users/kristoffer/.julia/dev/Pkg/src

(src) pkg>
```

What you might want is to have the same behavior as we had before where you try to find the project based on your current path but we wanted to write that as `@..` or something?

Reply:
Could just change this to `project === nothing ? nothing : abspath(project)` to fix Stefans comment?

Reply:
There's no way to do that from the command-line or environment, but you can accomplish it now by doing `Base.HOME_PROJECT[] = "@."` to circumvent the startup time expansion and delay it until load time.